### PR TITLE
do not finalize PETSc/SLEPc if not initalized from deal.II

### PR DESCRIPTION
**Pull request by stefanozampini**
@luca-heltai @bangerth This should be quick :-)

### CODE REVIEW

**Review by luca-heltai — APPROVED** [review]

File: source/base/mpi.cc [hunk 1]
```diff
@@ -778,11 +778,13 @@ namespace Utilities
 #ifdef DEAL_II_WITH_PETSC
 #  ifdef DEAL_II_WITH_SLEPC
       // Initialize SLEPc (with PETSc):
-      ierr = SlepcInitialize(&argc, &argv, nullptr, nullptr);
+      finalize_petscslepc = SlepcInitializeCalled ? false : true;
+      ierr                = SlepcInitialize(&argc, &argv, nullptr, nullptr);
```
Comment (tjhei) [inline source/base/mpi.cc:782]: Why do you initialize here even though it is already initialized?

Comment (masterleinad) [reply source/base/mpi.cc:782]: Looks like it's idempotent, https://gitlab.com/petsc/petsc/-/blob/main/src/sys/objects/pinit.c#L1230-1236. Subsequent calls are ignored after `PetscInitialized` has been called once.

Comment (stefanozampini) [reply source/base/mpi.cc:782]: yes

**Comment by stefanozampini** [comment]
@tjhei @masterleinad This can be merged

**Pull request merged**